worlds/hk/__init__.py

```diff
@@ -509,9 +509,13 @@ def set_goal(player, grub_rule: typing.Callable[[CollectionState], bool]):
                     per_player_grubs_per_player[player][player] += 1
 
                 if grub.location and grub.location.player in group_lookup.keys():
-                    for real_player in group_lookup[grub.location.player]:
+                    # will count the item linked grub instead
+                    pass
+                elif player in group_lookup:
+                    for real_player in group_lookup[player]:
                         grub_count_per_player[real_player] += 1
                 else:
+                    # for worlds not linking grubs but in a group
```

Comment:
This conditional branch is hit by all non-linked and non-group grubs. Typically that would just be players that are not in a group that is linking grubs, but even when grubs are linked, it also includes the grubs from item links with `replacement_item: Grub` and `link_replacement: false`, and item plando-ed `from_pool: false` grubs.

```suggestion
                    # for non-linked grubs
```